modules/components/src/utils/download.js

```diff
@@ -43,12 +43,19 @@ function createIFrame({ method, url, fields }) {
   return iFrame;
 }
 
-function download({ url, params, method = 'GET', body = {} }) {
+function download({
+  url,
+  params,
+  method = 'GET',
+  body = {},
+  callback = () => {},
```

Comment:
the name `callback` here could be a bit confusing. `onDownloadStart` or end? or whatever

Reply:
Probably don't need the callback option here, saveTSV returns download() which returns the promise so it could just be `saveTSV({...}).then((downloadKey) => onTSVExported({ downloadKey, files }))` would just need to resolve with `downloadKey`

Reply:
hehe i should've scrolled down. the diff shows the promise and I kinda figured but didn't look